app/controllers/user.js

```diff
@@ -34,3 +37,41 @@ exports.singUp = (req, res, next) => {
     next(err);
   }
 };
+
+exports.singIn = (req, res, next) => {
+  const user = req.body
+    ? {
+        email: req.body.email,
+        password: req.body.password
+      }
+    : {};
+
+  const signErrors = validateUser(user);
+
+  try {
+    if (!signErrors.valid) throw errors.signInError(signErrors.messages);
+
+    return User.getUserByEmailAndPassword(user.email)
+      .then(result => {
+        if (!result) throw errors.signInError(['user not registered']);
```

Comment:
Send directly the message not as an array (check this in all errors in this function)